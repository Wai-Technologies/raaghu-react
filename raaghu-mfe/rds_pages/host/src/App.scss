//@import "raaghu-react-themes/build/styles/default.css";
// Please dont remove below comment line Adding raaghu-theme in local side.
@import "../../../../raaghu-react-themes/src/styles/default.scss";

.main-body {
    margin-top: 29px;
    position: relative;
}
