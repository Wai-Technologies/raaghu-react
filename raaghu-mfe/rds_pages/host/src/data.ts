import { useTranslation } from "react-i18next";

export const getTranslatedThemeItems = () => {
  const { t } = useTranslation();

  const themeItems = [
    {
      id: 0,
      label: t("Light"),
      val: "Light",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/sun.json",
    },
    {
      id: 1,
      label: t("Dark"),
      val: "Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/moon.json",
    },
    {
      id: 2,
      label: t("Semi Dark"),
      val: "Semi Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/semi-dark.json",
    },
  ];

  return themeItems;
};

export const formsChildren = [
  {
    key: "5-0",
    label: "View",
    path: "/forms-view/:id",
    permission: "Forms.Form",
  },
];

export const getLabelForPath: any = (path: string, navItems: any) => {
  let label = null;
  for (const navItem of navItems) {
    if (navItem.path === path) {
      return navItem.label;
    }
    if (navItem.children) {
      label = getLabelForPath(path, navItem.children);
      if (label) {
        return label;
      }
    }
  }
  return label;
};