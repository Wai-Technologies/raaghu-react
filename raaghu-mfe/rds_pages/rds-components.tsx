export * from "../../raaghu-components/src/index";
