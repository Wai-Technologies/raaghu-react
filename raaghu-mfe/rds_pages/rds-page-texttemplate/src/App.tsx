import React, { Suspense } from "react";
import TextTemplate from "./texttemplate/text-template";

const App = () => (
    <Suspense>
        <TextTemplate></TextTemplate>
    </Suspense>
);

export default App;
