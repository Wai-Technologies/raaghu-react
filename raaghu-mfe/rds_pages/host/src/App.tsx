import React, { Suspense } from "react";
import "bootstrap/dist/js/bootstrap.bundle.min.js";
import "./App.scss";
import AuthHigherComponent from "./AuthHigherComponent";

const App = () => {
    return (
        <Suspense>
            <AuthHigherComponent></AuthHigherComponent>
        </Suspense>
    );
};

export default App;
