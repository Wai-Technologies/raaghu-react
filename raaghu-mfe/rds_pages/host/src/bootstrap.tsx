import App from "./App";
import ReactDOM from "react-dom/client";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../../libs/public.api";
import { Provider } from "react-redux";

(ReactDOM as any).createRoot(document.getElementById("root")).render(
    <Provider store={store}>
        <div>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </div>
    </Provider>
);

