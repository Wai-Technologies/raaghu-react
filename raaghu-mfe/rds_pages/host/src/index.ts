import("./bootstrap");
import "../../../web.config";
