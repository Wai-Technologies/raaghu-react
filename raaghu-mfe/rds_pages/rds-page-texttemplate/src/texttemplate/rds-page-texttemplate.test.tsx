import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../../libs/state-management";
import TextTemplate from "./text-template";
import { RdsCompDatatable } from "../../../rds-components";
import { RdsButton, RdsLabel, RdsOffcanvas, RdsSelectList, RdsTextArea } from "../../../../../raaghu-elements/src";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    initReactI18next: { // add this line
      type: "3rdParty",
      init: () => {},
    },
  }));
jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
  } as any;
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
  }));
  jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
  }));
  jest.mock("../../../../libs/state-management/index.ts", () => ({
    store: {
      getState: jest.fn().mockReturnValue({
        persistedReducer: {
          configureStore: {
            configuration: {
              auth: {
                grantedPolicies: {},
              },
            },
          },
        },
      }),
      dispatch: jest.fn(),
      subscribe: jest.fn(),
    },
  }));
  
  // Mock the rds-elements and rds-components
  jest.mock("../../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsOffcanvas: jest.fn(),
    RdsAlert: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsLabel : jest.fn(),
    RdsInput : jest.fn(),
    RdsIcon : jest.fn(),
    RdsCheckbox : jest.fn(),
    RdsSelectList : jest.fn(),
    RdsBadge : jest.fn(),
    RdsFabMenu : jest.fn(),
    RdsNavtabs : jest.fn(),
    RdsDatePicker : jest.fn(),
    RdsDropdownList : jest.fn(),
    RdsTextArea : jest.fn(),
  })); 
  
  jest.mock("../../../../../raaghu-components/src", () => ({
    RdsCompApiScopeBasicResource: jest.fn(),
    RdsCompDatatable: jest.fn(),
    RdsCompAlertPopup: jest.fn(),
    RdsOffcanvas:jest.fn(),
    RdsCompForgotPassword: jest.fn(),
    RdsCompFormsEmail: jest.fn(),
    RdsCompIconList: jest.fn(),
    RdsCompNewLanguage: jest.fn(),
    RdsCompWebsiteLog: jest.fn(),
    RdsCompCache: jest.fn(),
    RdsCompIdentityManagement: jest.fn(),
    RdsCompEmail : jest.fn(),
    RdsCompAccount : jest.fn(),
    RdsCompFeatures : jest.fn(),
    RdsCompIdentityOauthManagement  : jest.fn(),
    RdsCompTenantInformation : jest.fn(),
    RdsCompDatabaseConnection : jest.fn(),
    RdsCompSetPassword : jest.fn(),
  }));
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: jest.fn(),
  }));
  
  // Mock localStorage in your test setup
  const localStorageMock: Storage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
    length: 0, // Add length property
    key: jest.fn(), // Add key method
  };
  
  global.localStorage = localStorageMock;
  describe("texttemplate Page", () => {
it("renders texttemplate page correctly", () => {
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
        <Provider store={store}>
            <TextTemplate />
        </Provider>
    );
        document.querySelector('[label="TextTemplateManagement.Name"]');
        document.querySelector('[label="TextTemplateManagement.TemplateContent"]');
        document.querySelector('[button="AbpUi.Cancel"]');
        document.querySelector('[button="TextTemplateManagement.RestoreToDefault"]');
        document.querySelector('[button="AbpUi.Save"]');

});

it("Check texttemplate page with functionalty", () => {
  const tableHeaders = {
    key: "id",
    label: "TextTemplateManagement.Id",
    sortable: true,
  };
  const tableData = {
    id: "1",
    name: "test",
    templateContent: "test",
  };
  const onActionSelection = jest.fn();
  const paginationHandler = jest.fn();
  const setReferenceContent = jest.fn();
  const restoreDefaultContent = jest.fn();
  const onCloseFn = jest.fn();
  const handlerEditSave = jest.fn();
  const languages = {
    value: "en",
    label: "English",
    option:"option1"
  };
  const handleRefChange = jest.fn();
  const setTargetContent = jest.fn();
  const restoreDefault = jest.fn();
  const handlerCustomise = jest.fn();
  const mockDispatch = jest.fn();
  (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
 const text = render(
      <Provider store={store}>
          <TextTemplate />
          <RdsCompDatatable tableHeaders={[{ ...tableHeaders, displayName: "", datatype: "" }]} tableData={[tableData]} pagination={true} onActionSelection={onActionSelection}  onPaginationHandler={paginationHandler}/>
          <RdsOffcanvas placement={"end"} backDrop={true} scrolling={false} offId={"text-temp-edit-off"} canvasTitle={"TextTemplateManagement.Contents"} />
          <RdsLabel label="TextTemplateManagement.Name"/>
          <RdsTextArea placeholder={""} onChange={(e: any) => setReferenceContent(e.target.value)} rows={10}/>
          <RdsButton label="TextTemplateManagement.RestoreToDefault"  onClick={restoreDefaultContent}/>
          <RdsButton label="AbpUi.Cancel"  onClick={(e: any) => onCloseFn(e)} databsdismiss="offcanvas"/>
          <RdsButton label="AbpUi.Save" onClick={(e: any) => handlerEditSave(e)}/>
          <RdsOffcanvas placement={"end"} backDrop={false} scrolling={false} offId={"cust-per-cult"} canvasTitle={"TextTemplateManagement.CustomizePerCulture"} />
          <RdsLabel label="TextTemplateManagement.Name"/>
          <RdsSelectList selectItems={[languages]} id={"text.base"} label="TextTemplateManagement.BaseCultureName"  onChange={(item: any) => handleRefChange(item.value, "ref", "selectedTargetCulture")}/>
          <RdsTextArea placeholder={""}  onChange={(e: any) => setReferenceContent(e.target.value)} label="TextTemplateManagement.BaseContent"/>
          <RdsSelectList selectItems={[languages]} id={"baseTe"} label="extTemplateManagement.TargetCultureName" onChange={(item: any) => handleRefChange(item.value, "target", "selectedTargetCulture")}/>
          <RdsTextArea placeholder={""} onChange={(e: any) => setTargetContent(e.target.value)} label="TextTemplateManagement.TargetContent"/>
          <RdsButton label="TextTemplateManagement.RestoreToDefault"  onClick={restoreDefault}/>
          <RdsButton label="AbpUi.Cancel" onClick={(e: any) => onCloseFn(e)}/>
          <RdsButton label="AbpUi.Save" onClick={(e: any) => handlerCustomise(e)}/>
      </Provider>
  );
  expect(text).toMatchSnapshot();
});

afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
  })