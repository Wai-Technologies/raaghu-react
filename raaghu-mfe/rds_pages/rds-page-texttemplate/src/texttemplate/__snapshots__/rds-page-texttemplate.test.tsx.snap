// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`texttemplate Page Check texttemplate page with functionalty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid m-0 p-0"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
            >
              <div
                class="offset-md-4 col-md-4 mt-3 col-12 position-absolute position-lg-relative bottom-0 mb-3 custom-responsive-alert"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid m-0 p-0"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
          >
            <div
              class="offset-md-4 col-md-4 mt-3 col-12 position-absolute position-lg-relative bottom-0 mb-3 custom-responsive-alert"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
